### PR TITLE
Update uni v3.0.8 to v3.2.3

### PR DESCRIPTION
**Pull request by anxolin**
Updates and adapt to uniswap v3.2.3

## Tasks
- [x] Merge and resolve conflicts
- [x] Fix HeaderMod, it changed a bit. Main change is a theme toggle instead of the gear icon for the settings (which is in the widget now)
- [x] Make sure the new way of selecting lists don't break anything --> #107 
- [x] Review the "Default lists of lists" functionality. `DEFAULT_ACTIVE_LIST_URLS: string[] = [GEMINI_LIST]`
- [x] Check how routes are being displayed: SwapRoute use to have CurrencyLogo, not anymore:  import CurrencyLogo from '../CurrencyLogo'
- [x] Review new file imports, for STATE, SYLE, UTILS etc.
- [x] Handle new tags! It created branches with the new versions. We need a process to update
- [x] Fix unit tests #174 

Not in this PR:
- Review `serviceWorkerRegistration` #171 
- Carefully check the new commits #172 
- Blocked addresses #175
- Review analytics #176

## Overview

* The gear icon (settings), moved to the widget
* New theme toggle button in the header
image

* Improved token selector
image

* Improved list selection, now with combination of lists!
image

* Better support for custom tokens
image

* When a token is not found, but is in one of the lists. It suggest to add it
image

* Route is simplified
image

image

* Also, the confirmation modal has been approved

image

* Allows you to add the token to Metamask

image

image

* Expert mode don't makes you write "confirm" any more
image

* Allow to disable multi-hops:
image

image

## Notes on how to do the merge (notes for next merge)
* Make sure you have the `upstream` remote. If not ` git remote add upstream git@github.com:Uniswap/uniswap-interface.git`
* Fetch all info from the remote. 🚨 **IMPORTANT**: Don't bring their TAGSs 🚨: `git fetch upstream--no-tags`
* Make sure you know the latest version that was merged. In this case `v3.0.8`
* Make sure you know the latest version you want to merge. In this case `v3.2.3`
* Review and document the changes: `git log v3.0.8..v3.2.3 --oneline --decorate=1`
* Create a new branch for the merge: `git checkout -b update-uni-v3.2.3`
* Merge: `git merge v3.0.8` 
* Resolve conflicts: try to favour their changes in the original files. Theoretically most of it, should be imports that we've done as absolut while theirs are relative
* Try to run the app
* Review new files that where added in the merge. They might import modified files of our `custom` dir. So, they would require to change the import to use global. 
* Review commits for changes we could be affected, or how we can use the change. Get a good understanding of the new features, fixes, etc
* Review the Mod files. They should still compare line to line. So you can use vscode to compare the two files. Try to make it so they still compare line by line. 

* For making the tags , maybe next time there's a better way. But this time, I made the mistake of fetching all the tags from `upstream`, so it brought their versioning into ours. Anyways, I took the output, that showed the new branches and I created their new versions as versions in `uniswap/<version>`

```
\[new tag\] (.+) ->.+

git tag upstream/$1 uniswap/$1 && \
```

## Notes on the conflict solving
* Most of the conflicts, where import conflicts
* Some updates of dependencies
* Fix some conflicts with global imports instead of relative
* Deleted ListSeelct component
* Changed how the lists are handled
    * Affected the constants
    * For exaple affected at dimas state/orders/mocks.ts
        * changed useSelectedTokenList ---> useCombinedActiveList
        * Also changes what it returns
* Adapted a few things to xDAI
    * Like combination logic for tokenLists: `combineMaps` in `state/list/hooks`
* A few conflicts in custom!
* They added WBTC for route calculations
* Now the settings is in the widget
* Changed the HeaderMod: compared with a diff and adapted it
    * deleted thigns, added things :) 
* Darkmode!
* A lot of tags came when I updated. I had to
    * Save them in a file
    * Execute a command

## Output of the merge
```
❯ git merge v3.2.3
Auto-merging yarn.lock
CONFLICT (content): Merge conflict in yarn.lock
Auto-merging tsconfig.json
Auto-merging src/theme/index.tsx
Auto-merging src/state/user/hooks.tsx
Auto-merging src/state/mint/hooks.ts
Auto-merging src/state/lists/updater.ts
CONFLICT (content): Merge conflict in src/state/lists/updater.ts
Auto-merging src/state/lists/reducer.ts
CONFLICT (content): Merge conflict in src/state/lists/reducer.ts
Auto-merging src/state/lists/reducer.test.ts
CONFLICT (content): Merge conflict in src/state/lists/reducer.test.ts
Auto-merging src/state/lists/hooks.ts
CONFLICT (content): Merge conflict in src/state/lists/hooks.ts
Auto-merging src/state/governance/hooks.ts
Auto-merging src/pages/Vote/index.tsx
Auto-merging src/pages/Vote/VotePage.tsx
CONFLICT (content): Merge conflict in src/pages/Vote/VotePage.tsx
Auto-merging src/pages/Swap/index.tsx
CONFLICT (content): Merge conflict in src/pages/Swap/index.tsx
Auto-merging src/pages/RemoveLiquidity/index.tsx
Auto-merging src/pages/PoolFinder/index.tsx
Auto-merging src/pages/Pool/index.tsx
Auto-merging src/pages/MigrateV1/index.tsx
CONFLICT (content): Merge conflict in src/pages/MigrateV1/index.tsx
Auto-merging src/pages/Earn/index.tsx
Auto-merging src/pages/Earn/Manage.tsx
Auto-merging src/pages/App.tsx
Auto-merging src/pages/AddLiquidity/index.tsx
Auto-merging src/index.tsx
CONFLICT (content): Merge conflict in src/index.tsx
Auto-merging src/hooks/useFetchListCallback.ts
Auto-merging src/data/V1.ts
Auto-merging src/constants/index.ts
CONFLICT (content): Merge conflict in src/constants/index.ts
Auto-merging src/components/swap/SwapRoute.tsx
CONFLICT (content): Merge conflict in src/components/swap/SwapRoute.tsx
Auto-merging src/components/swap/AdvancedSwapDetails.tsx
Auto-merging src/components/earn/PoolCard.tsx
Auto-merging src/components/TransactionConfirmationModal/index.tsx
CONFLICT (content): Merge conflict in src/components/TransactionConfirmationModal/index.tsx
Auto-merging src/components/TokenWarningModal/index.tsx
CONFLICT (content): Merge conflict in src/components/TokenWarningModal/index.tsx
Auto-merging src/components/Settings/index.tsx
Auto-merging src/components/SearchModal/filtering.ts
CONFLICT (content): Merge conflict in src/components/SearchModal/filtering.ts
CONFLICT (modify/delete): src/components/SearchModal/ListSelect.tsx deleted in v3.2.3 and modified in HEAD. Version HEAD of src/components/SearchModal/ListSelect.tsx left in tree.
Auto-merging src/components/SearchModal/CurrencySearch.tsx
CONFLICT (content): Merge conflict in src/components/SearchModal/CurrencySearch.tsx
Auto-merging src/components/SearchModal/CurrencyList.tsx
CONFLICT (content): Merge conflict in src/components/SearchModal/CurrencyList.tsx
Auto-merging src/components/PositionCard/index.tsx
Auto-merging src/components/Popups/ListUpdatePopup.tsx
Auto-merging src/components/Menu/index.tsx
Auto-merging src/components/Header/index.tsx
Auto-merging src/components/CurrencyInputPanel/index.tsx
Auto-merging package.json
CONFLICT (content): Merge conflict in package.json
Removing cypress/integration/token-warning.ts
Auto-merging README.md
CONFLICT (content): Merge conflict in README.md
Automatic merge failed; fix conflicts and then commit the result.

```

## Changes
34dfb41 (tag: v3.2.3) feature(service worker): add offline support (#1319)
e77fcd2 (tag: v3.2.2) Update index.ts
0b7846e (tag: v3.2.1) fix(discord): correct links (#1315)
f450d34 (tag: v3.2.0) feat(transactions): enable button to add tokens to metamask  (#1311)
76ab349 show hidden search results by default (#1310)
5c3c1c6 run npx browserslist@latest --update-db (#1303)
9efd5da (tag: v3.1.11) change branch in workflows
8fd894f (upstream/master) fix a prominent instance (#1304)
cc22183 (tag: v3.1.10) fix(modal cutoff, lists): make modal scrollable on small screens, fix typo, auto update all lists (#1301)
9175dd1 remove cloudflare insights
bbd50f0 fixes https://github.com/Uniswap/uniswap-interface/issues/1214
2291e3e (tag: v3.1.9) improvement(lists): enable Gemini by default (#1276)
28d8f0b (tag: v3.1.8) Improvement(token import): update import flow and style tweaks (#1273)
8bed390 (tag: v3.1.7) Improvements(lists): update defaults, style updates (#1270)
a1000c6 (tag: v3.1.6) improvement(routing): use WBTC pairs for routing swaps, style updates (#1269)
267204d (tag: v3.1.5) Improvement(lists): Switch to multiple active lists (#1237)
74f50f1 (tag: v3.1.4) fix(typo): toggle (#1260)
a70aa41 (tag: v3.1.3) Routing updates only (#1265)
587b659 (tag: v3.1.2) catch error in bytes32 string parsing (#1253)
5388cab (tag: v3.1.1) update vote timestamp estimation (#1242)
cadd68f (tag: v3.1.0) feat: allow overriding proposals in the UI (#1239)
ab8ce37 (tag: v3.0.18) fix(token lists): update to latest token-lists
5a9a71a improvement(analytics): try out cloudflare analytics
b93fd23 (tag: v3.0.17) Specially Designated Nationals and Blocked Persons Ethereum addresses
e9a11bb (tag: v3.0.16) fix(vote): make the vote page responsive
c5afbed (tag: v3.0.15) fixed bug causing wbtc to have 18 decimals (#1211)
5b2c445 (tag: v3.0.14) fix(list popup): big token list updates scroll issue (#1219)
7fd4005 (tag: v3.0.13) Fix(UNI page): hide inactive pools (#1213)
48eab0d (tag: v3.0.12) bump token lists
c9ee1b3 (tag: v3.0.11) Fix(pool page): show deposited liquidity on pool page (#1195)
eb4c305 (tag: v3.0.10) fix(ens): support ens names with dashes in them
d982562 (tag: v3.0.9) Allow absolute imports (#1185)

## Tags
* Updated tags manually
* If you fetch directly from `upstream` you'll get some tags you don't want
*  One solution is to tag manually:
    *  Get all tags and commits from uni `git ls-remote --tags upstream > tags.txt`
    * Delete from the TXT file the tags you don't want. Include only the ones between the versions you are updating
    * Use vscode to find and replace using a regex:
       * expression: `(\w+)	refs/tags/(.+)`  
       * value: `git tag uniswap/$2 $1 && \`
    * Execute the command (remember to delete the last && of the command

In this case, the execution returned:
```
❯ git push --tags
Total 0 (delta 0), reused 0 (delta 0)
To github.com:gnosis/gp-swap-ui.git
 * [new tag]         uniswap/v3.0.9 -> uniswap/v3.0.9
 * [new tag]         uniswap/v3.1.0 -> uniswap/v3.1.0
 * [new tag]         uniswap/v3.1.1 -> uniswap/v3.1.1
 * [new tag]         uniswap/v3.1.10 -> uniswap/v3.1.10
 * [new tag]         uniswap/v3.1.11 -> uniswap/v3.1.11
 * [new tag]         uniswap/v3.1.2 -> uniswap/v3.1.2
 * [new tag]         uniswap/v3.1.3 -> uniswap/v3.1.3
 * [new tag]         uniswap/v3.1.4 -> uniswap/v3.1.4
 * [new tag]         uniswap/v3.1.5 -> uniswap/v3.1.5
 * [new tag]         uniswap/v3.1.6 -> uniswap/v3.1.6
 * [new tag]         uniswap/v3.1.7 -> uniswap/v3.1.7
 * [new tag]         uniswap/v3.1.8 -> uniswap/v3.1.8
 * [new tag]         uniswap/v3.1.9 -> uniswap/v3.1.9
 * [new tag]         uniswap/v3.2.0 -> uniswap/v3.2.0
 * [new tag]         uniswap/v3.2.1 -> uniswap/v3.2.1
 * [new tag]         uniswap/v3.2.2 -> uniswap/v3.2.2
 * [new tag]         uniswap/v3.2.3 -> uniswap/v3.2.3
 * [new tag]         upstream/v3.0.8 -> upstream/v3.0.8
```

# Test
Good Testing required with the review

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -19,10 +21,10 @@ export const USDT = new Token(ChainId.MAINNET, '0xdAC17F958D2ee523a2206206994597
 export const COMP = new Token(ChainId.MAINNET, '0xc00e94Cb662C3520282E6f5717214004A7f26888', 18, 'COMP', 'Compound')
 export const MKR = new Token(ChainId.MAINNET, '0x9f8F72aA9304c8B593d555F12eF6589cC3A579A2', 18, 'MKR', 'Maker')
 export const AMPL = new Token(ChainId.MAINNET, '0xD46bA6D942050d489DBd938a2C909A5d5039A161', 9, 'AMPL', 'Ampleforth')
-export const WBTC = new Token(ChainId.MAINNET, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599', 18, 'WBTC', 'Wrapped BTC')
+export const WBTC = new Token(ChainId.MAINNET, '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599', 8, 'WBTC', 'Wrapped BTC')
```
Comment (alfetopito) [inline src/constants/index.ts:24]: Oopsie

File: src/custom/components/Header/HeaderMod.tsx [hunk 2]
```diff
@@ -11,13 +11,13 @@ import Logo from 'assets/svg/logo.svg'
 import LogoDark from 'assets/svg/logo_white.svg'
 import { useActiveWeb3React } from 'hooks'
 import { useDarkModeManager } from 'state/user/hooks'
-import { useETHBalances } from 'state/wallet/hooks'
+import { useETHBalances /*, useAggregateUniBalance*/ } from 'state/wallet/hooks'
 // import { CardNoise } from 'components/earn/styled'
 // import { CountUp } from 'use-count-up'
-import { ExternalLink } from 'theme'
+import { /*TYPE,*/ ExternalLink } from 'theme'
```
Comment (alfetopito) [inline src/custom/components/Header/HeaderMod.tsx:17]: Any reason to keep the comment?

**Review by alfetopito — COMMENTED** [review]
Well, as far as I can tell, all is ok?
Integration tests are failing though.
And I have not tested it

File: src/custom/components/Header/HeaderMod.tsx (hunk 2 above)
Comment (anxolin) [reply src/custom/components/Header/HeaderMod.tsx:17]: Not a big one. I just wanted to keep the original code as much as possible. That was in it. Anyways, I'd be ok to remove it or leave it. Let me know what u prefer

Comment (alfetopito) [reply src/custom/components/Header/HeaderMod.tsx:17]: I don't mind

**Comment by anxolin** [comment]
@alfetopito @W3stside I did the review, and fixed the test. Tested a few things. There was a lot of changes. We should keep an eye, although all looks good. 

Let me know if you want to do further review of this PR. 

I think we should do frequent updates, so it doesn't get to this size of PRs. Maybe our biweekly deployment is a good time for making a PR of updating Uni.

**Pull request merged**